### PR TITLE
fix insert inode fail when load deleting partition

### PR DESCRIPTION
**Pull request by cw123**
### What problem does this PR solve?

Issue Number:  #995

Problem Summary: fix insert inode fail when load deleting partition

### What is changed and how it works?

What's Changed:

How it Works:

Side effects(Breaking backward compatibility? Performance regression?): 

### Check List

- [ ] Relevant documentation/comments is changed or added
- [ ] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

**Review by xu-chaojie — APPROVED** [review]

File: curvefs/src/metaserver/metastore.cpp [hunk 1]
```diff
@@ -69,9 +60,10 @@ bool MetaStoreImpl::LoadInode(uint32_t partitionId, void* entry) {
     }
 
     auto inode = reinterpret_cast<Inode*>(entry);
-    auto rc = partition->InsertInode(*inode);
+    MetaStatusCode rc = partition->InsertInode(*inode);
     if (rc != MetaStatusCode::OK) {
-        LOG(ERROR) << "InsertInode failed, retCode = " << rc;
+        LOG(ERROR) << "InsertInode failed, retCode = "
```
Comment (xu-chaojie) [inline curvefs/src/metaserver/metastore.cpp:65]: using “：” instead of “=” in log

File: curvefs/src/metaserver/metastore.cpp [hunk 2]
```diff
@@ -137,6 +130,20 @@ bool MetaStoreImpl::Load(const std::string& pathname) {
         partitionMap_.clear();
         LOG(ERROR) << "Load metadata failed.";
     }
+
+    for (auto it = partitionMap_.begin(); it != partitionMap_.end(); it++) {
```
Comment (wu-hanqing) [inline curvefs/src/metaserver/metastore.cpp:134]: use range-based for loop

**Review by wu-hanqing — COMMENTED** [review]
you should add tests to verify this patch.

File: curvefs/src/metaserver/metastore.cpp (hunk 2 above)
Comment (cw123) [reply curvefs/src/metaserver/metastore.cpp:134]: The two methods are equivalent

**Comment by cw123** [comment]
> you should add tests to verify this patch.

done, add test case

File: curvefs/test/metaserver/metastore_test.cpp [hunk 3]
```diff
@@ -94,7 +94,10 @@ class MetastoreTest : public ::testing::Test {
                    first.partitionid() == second.partitionid() &&
                    first.start() == second.start() &&
                    first.end() == second.end() &&
-                   first.nextid() == second.nextid();
+                   first.nextid() == second.nextid() &&
```
Comment (wu-hanqing) [inline curvefs/test/metaserver/metastore_test.cpp:97]: you can use `google::protobuf::util::MessageDifferencer::Equals` to simplify this compare

**Review by wu-hanqing — APPROVED** [review]

**Comment by cw123** [comment]
recheck

**Comment by cw123** [comment]
recheck

**Comment by cw123** [comment]
recheck

**Pull request merged**